Pcap++/src/PcapLiveDevice.cpp

```diff
@@ -488,29 +518,44 @@ int PcapLiveDevice::startCaptureBlockingMode(OnPacketArrivesStopBlocking onPacke
 	m_cbOnPacketArrivesBlockingMode = onPacketArrives;
 	m_cbOnPacketArrivesBlockingModeUserCookie = userCookie;
 
-	long startTimeSec = 0, startTimeNSec = 0;
-	clockGetTime(startTimeSec, startTimeNSec);
-
-	long curTimeSec = 0;
+	auto startTime = std::chrono::steady_clock::now();
 
+	auto currentTime = startTime;
 	m_CaptureThreadStarted = true;
 	m_StopThread = false;
 
-	if (timeout <= 0)
+	struct pollfd pcapPollFd;
+	memset(&pcapPollFd, 0, sizeof(pcapPollFd));
+	pcapPollFd.fd = m_PcapSelectableFd;
+	pcapPollFd.events = POLLIN;
+
+	constexpr int shortIntervalMs = 10;
+
+	while (!m_StopThread)
 	{
-		while (!m_StopThread)
+		if(m_usePoll)
 		{
-			pcap_dispatch(m_PcapDescriptor, -1, onPacketArrivesBlockingMode, (uint8_t*)this);
+			int ready = poll(&pcapPollFd, 1, shortIntervalMs); // wait for few milliseconds
```

Comment:
Why do we use `shortIntervalMs` instead of `timeout`? Isn't the whole point is to use `poll` for the timeout provided by the user?

Reply:
the logic is like this:
```
while(not timeout) {
     if(check if poll got a packet "in" event within a short period) {
        pcap_function()
     }
     
     // either poll got packets or poll got timeout , we can check if the while got timeout
}
```

ok, now imagine a case, the while loop timeout is 2000ms, and we set the poll timeout also 2000ms.
0. start time is 0 ms.
1. a packet comes at time 1000ms
2. while-loop not timeout, go to the next loop
3. no packets coming anymore
4. poll waits for 2000ms
And you can see that the whole process causes 1000+2000 ms

Reply:
You're right again, I forgot what the meaning of `timeout` was 🤦‍♂️ 

But maybe we can build this logic which might be more accurate:
```
startTime = now
pollTimeout = timeout
while (not timeout) {
    poll(pollTimeout)

    if (poll timed out) {
        break and return -1
    }

    pcap()

    pollTimeout -= now - startTime
}
```

That should solve your scenario:

0. start time is 0 ms. poll timeout is 2000ms
1. a packet comes at time 1000ms
2. while-loop not timeout, go to the next loop. poll timeout is now 1000ms
3. no packets coming anymore
4. poll waits for 1000ms and quits the loop

Reply:
That sounds good, changed.